File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/RefreshVideoListUseCaseImpl.java
package com.hwonchul.movie.domain.usecase.details;

import androidx.annotation.NonNull;

import com.hwonchul.movie.domain.repository.VideoRepository;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Completable;

public class RefreshVideoListUseCaseImpl implements RefreshVideoListUseCase {
    private final VideoRepository repository;

    @Inject
    public RefreshVideoListUseCaseImpl(@NonNull VideoRepository repository) {
        this.repository = repository;
    }

    @Override
    public Completable invoke(int movieId) {
        return repository.refreshFromRemote(movieId);
    }
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/RefreshVideoListUseCaseImpl.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.repository.VideoRepository
import javax.inject.Inject

class RefreshVideoListUseCaseImpl @Inject constructor(
    private val repository: VideoRepository
) : RefreshVideoListUseCase {

    override suspend fun invoke(movieId: Int): Result<Unit> =
        runCatching { repository.refreshFromRemote(movieId) }
}